### ISSUE TITLE
Doxygen 1.10.0: Regression when parsing template code

### ISSUE THREAD

**Issue by pkl97** [doxygen/doxygen#10593]
The attached test case triggers this warning which indicates that Doxygen misinterprets the code:
```
Test.hpp:19: warning: documented symbol 'A::std::enable_if_t(const U &p_value) const' was not declared or defined.
```

**Expected behavior**
No warning should be triggered as in Doxygen 1.9.8

**To Reproduce**
Run Doxygen 1.10.0 with the Doxyfile from the attachment

**Version**
Doxygen 1.10.0

[Doxygen-TestCase.tar.gz](https://github.com/doxygen/doxygen/files/14056506/Doxygen-TestCase.tar.gz)

**Comment by albert-github** [comment]
I tried with:
- 1.9.8 version warning
- 1.10.0 version warning
- current master (1.11.0 (51326cfc072bf6fe4ca13c60ca0e9faec1d38f82)) no warning.


The HTML output for the 1.9.8 version and the current master look identical.

I think the problem is fixed but not released.

**Comment by pkl97** [comment]
Interesting. I can confirm that the current 1.11.0 Doxygen trunk (51326cfc072bf6fe4ca13c60ca0e9faec1d38f82) does no longer show the warning (should have checked that before).

But at least in my setup 1.9.8 also does not show the warning.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
